File: src/main/java/com/thebluehats/server/game/utils/LoreParserBase.java
package com.thebluehats.server.game.utils;

import java.util.ArrayList;

import org.apache.commons.lang.StringUtils;
import org.bukkit.ChatColor;

public abstract class LoreParserBase<T> {
    protected final String lore;

    protected LoreParserBase(String lore) {
        this.lore = lore;
    }

    protected ArrayList<String> parseLore(String lore) {
        ArrayList<String> parsedLore = new ArrayList<>();

        String[] lines = lore.split("<br/>");

        for (String line : lines) {
            String formattedLine = ChatColor.GRAY.toString() + line.trim();

            for (ChatColor chatColor : ChatColor.values()) {
                String name = chatColor.name().toLowerCase();

                formattedLine = StringUtils.replace(formattedLine, "<" + name + ">", chatColor.toString());
                formattedLine = StringUtils.replace(formattedLine, "</" + name + ">", ChatColor.GRAY.toString());
            }

            formattedLine = insertVariableValuesForLine(formattedLine);

            parsedLore.add(formattedLine);
        }

        return parsedLore;
    }

    public abstract void setVariables(T variables);

    protected abstract String insertVariableValuesForLine(String line);
}

File: src/main/kotlin/com/thebluehats/server/game/utils/LoreParserBase.kt
package com.thebluehats.server.game.utils

import org.apache.commons.lang.StringUtils
import org.bukkit.ChatColor
import java.util.*

abstract class LoreParserBase protected constructor(protected val lore: String) {
    protected fun parseLore(lore: String): ArrayList<String> {
        val parsedLore = ArrayList<String>()
        val lines = lore.split("<br/>").toTypedArray()

        lines.forEach { l ->
            var formattedLine = insertVariableValuesForLine(ChatColor.GRAY.toString() + l)

            ChatColor.values().forEach { c ->
                val name = c.name.toLowerCase()

                formattedLine = StringUtils.replace(formattedLine, "-","_" )
                formattedLine = StringUtils.replace(formattedLine, "<$name>", c.toString())
                formattedLine = StringUtils.replace(formattedLine, "</$name>", ChatColor.GRAY.toString())
            }
            
            parsedLore.add(formattedLine)
        }

        return parsedLore
    }

    protected abstract fun insertVariableValuesForLine(line: String): String
}